java/flight/flight-grpc/src/main/java/org/apache/arrow/flight/FlightGrpcUtils.java

```diff
@@ -17,19 +17,111 @@
 
 package org.apache.arrow.flight;
 
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
 
 import org.apache.arrow.flight.auth.ServerAuthHandler;
 import org.apache.arrow.memory.BufferAllocator;
 
 import io.grpc.BindableService;
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
+import io.grpc.ConnectivityState;
 import io.grpc.ManagedChannel;
+import io.grpc.MethodDescriptor;
 
 /**
  * Exposes Flight GRPC service & client.
  */
 public class FlightGrpcUtils {
+  /**
+   * Proxy class for ManagedChannel that makes closure a no-op.
+   */
+  static class ProxyManagedChannel extends ManagedChannel {
+    private final ManagedChannel channel;
+    private boolean isShutdown;
+    private final List<Runnable> shutdownCallbacks;
+
+    ProxyManagedChannel(ManagedChannel channel) {
+      this.channel = channel;
+      this.isShutdown = channel.isShutdown();
+      this.shutdownCallbacks = new ArrayList<>();
+    }
+
+    @Override
+    public ManagedChannel shutdown() {
+      isShutdown = true;
+      shutdownCallbacks.forEach(c -> c.run());
+      return this;
+    }
+
+    @Override
+    public boolean isShutdown() {
+      return isShutdown;
+    }
+
+    @Override
+    public boolean isTerminated() {
+      return isShutdown;
+    }
+
+    @Override
+    public ManagedChannel shutdownNow() {
+      return shutdown();
+    }
+
+    @Override
+    public boolean awaitTermination(long l, TimeUnit timeUnit) {
+      return true;
```

Comment:
Are you allowed to call awaitTermination() before shutdown() (eg one thread awaits another thread to shutdown?)

Reply:
Perhaps this should actually simply be `return isShutDown;` - not sure we want to imitate other state.

Reply:
I don't think this is enough -- looking at the implementation in ManagedChannelImpl it just immediately blocks on a countdown latch until the timeout.

You could call awaitTermination first then call shutdown from another thread and then have awaitTermination return true.